yt_dlp/extractor/musicdex.py

```diff
@@ -0,0 +1,164 @@
+        }
+
+
+class MusicdexSongIE(MusicdexBaseIE):
+    _VALID_URL = r'https?://(?:www\.)?musicdex\.org/track/(?P<id>\d+)'
+
+    _TESTS = [{
+        'url': 'https://www.musicdex.org/track/306/dual-existence',
+        'info_dict': {
+            'id': '306',
+            'ext': 'mp3',
+            'title': 'dual existence',
+            'description': '#NIPPONSEI @ IRC.RIZON.NET',
+            'track': 'dual existence',
+            'track_number': 1,
+            'duration': 266000,
+            'genre': ['Anime'],
+            'like_count': int,
+            'view_count': int,
+            'artist': ['fripSide'],
+            'album_artist': ['fripSide'],
+            'thumbnail': 'https://www.musicdex.org/storage/album/9iDIam1DHTVqUG4UclFIEq1WAFGXfPW4y0TtZa91.png',
+            'album': 'To Aru Kagaku no Railgun T OP2 Single - dual existence',
+            'release_year': 2020
+        },
+        'params': {'skip_download': True}
+    }]
+
+    def _real_extract(self, url):
+        id = self._match_id(url)
+        data_json = self._download_json(f'https://www.musicdex.org/secure/tracks/{id}?defaultRelations=true', id)['track']
+        return self._return_info(data_json, data_json.get('album') or {}, id)
+
+
+class MusicdexAlbumIE(MusicdexBaseIE):
+    _VALID_URL = r'https?://(?:www\.)?musicdex\.org/album/(?P<id>\d+)'
+
+    _TESTS = [{
+        'url': 'https://www.musicdex.org/album/56/tenmon-and-eiichiro-yanagi-minori/ef-a-tale-of-memories-original-soundtrack-2-fortissimo',
+        'playlist_mincount': 28,
+        'info_dict': {
+            'id': '56',
+            'genre': ['OST'],
+            'view_count': int,
+            'artist': ['TENMON & Eiichiro Yanagi / minori'],
+            'title': 'ef - a tale of memories Original Soundtrack 2 ~fortissimo~',
+            'release_year': 2008,
+            'thumbnail': 'https://www.musicdex.org/storage/album/2rSHkyYBYfB7sbvElpEyTMcUn6toY7AohOgJuDlE.jpg',
+        },
+    }]
+
+    def _real_extract(self, url):
+        id = self._match_id(url)
+        data_json = self._download_json(f'https://www.musicdex.org/secure/albums/{id}?defaultRelations=true', id)['album']
+        entries = [self._return_info(track, data_json, track['id']) for track in data_json.get('tracks') or [] if track.get('id')]
+
+        return {
+            '_type': 'playlist',
+            'id': id,
+            'title': data_json.get('name'),
+            'description': data_json.get('description'),
+            'genre': [genre.get('name') for genre in data_json.get('genres') or []],
+            'view_count': data_json.get('plays'),
+            'artist': [artist.get('name') for artist in data_json.get('artists') or []],
+            'thumbnail': format_field(data_json, 'image', 'https://www.musicdex.org/%s'),
+            'release_year': try_get(data_json, lambda x: date_from_str(unified_strdate(x['release_date'])).year),
+            'entries': entries,
+        }
+
+
+class MusicdexPageIE(InfoExtractor):
```

Comment:
can't this be:
```suggestion
class MusicdexPageIE(MusicdexBaseIE):
```

Reply:
No conflicts right now but both classes server different purpose...I don't think combining them is the right approach.

Reply:
Actually the entries function can be added to the Base class itself, But i didn't do it for the same reason.

Reply:
Okay, just wondering since `_entries` currently seems to get data that can be directly parsed as playlist entries (like on `TikTokBaseListIE` or `GameJoltPostListBaseIE`)

Reply:
>can be directly parsed as playlist entries

The playlist and artist entries require diff processing and hence can't be hardcoded in the `MusicdexPageIE`
Basically artists request provides the albums and playlist request provides the tracks.
so this will be diff for playlists `self._return_info(track, album, track['id']) for track in album.get('tracks') or [] if track.get('id')`